### PR TITLE
fix: enhance plugin management with interactive removal prompts

### PR DESCRIPTION
**Pull request by elibosley**
- Add RemovePluginQuestionSet for interactive plugin removal
- Update plugin commands to use PluginManagementService
- Improve plugin installation error handling and warnings
- Clean up test fixtures and update plugin command tests
- Reset dev config to clean state (v4.11.0, no plugins)

## Summary by CodeRabbit

* **New Features**
  * Improved plugin management in the CLI with interactive prompts for plugin removal and enhanced error handling.
  * CLI plugin commands now provide clearer user feedback and warnings for missing plugins.
  * Added log suppression capability and dedicated plugin log file support.

* **Refactor**
  * Plugin CLI commands now use dedicated management services and interactive prompts instead of direct GraphQL operations, streamlining workflows and improving reliability.
  * Simplified CLI imports and logging for more straightforward error handling.
  * Deferred plugin module logging to application bootstrap for improved lifecycle management.
  * Updated logging service to respect global log suppression and added unconditional logging method.

* **Tests**
  * Refactored plugin CLI command tests for better isolation and coverage, using service mocks and enhanced prompt simulations.
  * Updated report command tests to reflect new logging behavior.

* **Chores**
  * Updated API configuration settings and removed outdated test fixture files and timestamp data.
  * Simplified test file logic by removing remote file download and cache functionality.
  * Adjusted build configuration to conditionally set CLI shebang based on environment.
  * Enhanced configuration file handler to optionally accept external logging.
  * Updated CLI command script to set environment variable for testing.
  * Added environment variables for log file paths and log suppression.
  * Updated logging setup to conditionally suppress logs and write plugin logs to file.
  * Improved error and output logging consistency across CLI commands.
  * Added placeholder file to ensure log directory version control tracking.

### CODE REVIEW

File: api/src/core/log.ts [hunk 1]
```diff
@@ -1,26 +1,38 @@
...
 export const levels = ['trace', 'debug', 'info', 'warn', 'error', 'fatal'] as const;
 
 export type LogLevel = (typeof levels)[number];
 
 const level = levels[levels.indexOf(LOG_LEVEL.toLowerCase() as LogLevel)] ?? 'info';
 
-export const logDestination = pino.destination();
+const nullDestination = pino.destination({
+    write() {
+        // Suppress all logs
```
Comment (elibosley) [inline api/src/core/log.ts:14]: Good idea.

File: api/src/unraid-api/cli/log.service.ts [hunk 2]
```diff
@@ -49,6 +55,11 @@ export class LogService {
         }
     }
 
+    always(...messages: unknown[]): void {
```
Comment (elibosley) [inline api/src/unraid-api/cli/log.service.ts:71]: Dope, great idea.

File: api/src/cli.ts [hunk 3]
```diff
@@ -1,29 +1,37 @@
...
         return shellToUse.stdout.trim();
     } catch (err) {
-        logger.debug('Could not find unraid-api in PATH, using default location');
-
         return '/usr/bin/unraid-api';
     }
 };
 
+const getLogger = () => {
+    if (LOG_LEVEL === 'TRACE' && !SUPPRESS_LOGS) {
```
Comment (elibosley) [inline api/src/cli.ts:21]: This seems smart.

File: api/vite.config.ts [hunk 4]
```diff
@@ -110,6 +110,9 @@ export default defineConfig(({ mode }): ViteUserConfig => {
                     interop: 'auto',
                     banner: (chunk) => {
                         if (chunk.fileName === 'main.js' || chunk.fileName === 'cli.js') {
+                            if (process.env.COMMAND_TESTER) {
```
Comment (elibosley) [inline api/vite.config.ts:113]: I made this change @pujitm  - should we just change the shebang to always use /usr/bin/env? Would be 100% reliable unless you blow up your path on Unraid...

Comment (pujitm) [reply api/vite.config.ts:113]: let's test that first -- just want to try some edge cases like rebooting, non-interactive ssh before including it in a release. maybe we can push it to the next release since it isn't critical and the current one is?
